lib/gollum/public/gollum/javascript/gollum.js.erb

```diff
@@ -143,23 +143,51 @@ function flashNotice(type, notice, button_label, button_function, button_type) {
   if (type == 'success') { setTimeout(function() {$('#gollum-flash').fadeOut();}, 5000); }
 }
 
+// Check whether the user has allowed their browser to access their system
+// clipboard.
+async function isClipboardAccessDenied() {
+  // Firefox doesn't support the "clipboard-write" permission but still allows
+  // websites to use the `Clipboard.writeText()` method unless the user has
+  // explicitly turned off Firefox clipboard access globally. Since we use
+  // `Clipboard.writeText()` we can return early for browsers that report as
+  // Firefox.
+  if (navigator.userAgent.indexOf("Firefox") > 0) {
```

Comment:
FWIW, in Firefox:

* `typeof(navigator.clipboard.writeText)` appears to always return `function`
* `navigator.clipboard.writeText("clipboard test")` works depending on whether the user has recently interacted with the site ([transient user activation](https://developer.mozilla.org/en-US/docs/Web/Security/User_activation)) 
* if the above requirement is not met, attempting to write to the clipboard will throw: `Promise { <state>: "rejected", <reason>: DOMException } Uncaught (in promise) DOMException: Clipboard write was blocked due to lack of user activation.`
* regardless of whether the transient user activation requirement is met, `navigator.permissions.query({name: "clipboard-write"})` will always throw an exception and return `undefined`: `Uncaught (in promise) TypeError: 'clipboard-write' (value of 'name' member of PermissionDescriptor) is not a valid value for enumeration PermissionName.`

Reply:
So instead of testing for the presence of 'Firefox' in the UA, we could perhaps test whether the permission check returns `undefined` -- and in that case assume that access to the clipboard is not denied. But given the complexities, I'm also happy to just go with checking the UA (provided checking for "Firefox" is enough to cover also other Gecko-based browsers that have the same behavior with regard to permissions).

Reply:
The reasons I did it this way initially is because the Firefox browser really is the exception here. But I've found another way to do this, inspired by @dometto's notes above. Let me know if you can think of any improvements:

```diff
 // Check whether the user has allowed their browser to access their system
 // clipboard.
 async function isClipboardAccessDenied() {
-  // Firefox doesn't support the "clipboard-write" permission but still allows
-  // websites to use the `Clipboard.writeText()` method unless the user has
-  // explicitly turned off Firefox clipboard access globally. Since we use
-  // `Clipboard.writeText()` we can return early for browsers that report as
-  // Firefox.
-  if (navigator.userAgent.indexOf("Firefox") > 0) {
-    return false;
-  }
-
   if (navigator.permissions && navigator.permissions.query) {
     return await navigator.permissions.query({name: "clipboard-write"})
       .then((permissionStatus) => permissionStatus.state === "denied")
-      .catch((error) => console.error(error));
+      .catch((error) => {
+
+        // If the error is about a browser not supporting the 'clipboard-write'
+        // PermissionDescriptor member, then we can safely return false, as the
+        // browser should not require clipboard permissions for copy-to-clipboard
+        // functionality. Of the major browsers, only Firefox should encounter
+        // an error as of this writing.
+        //
+        // Here's Firefox's current error message for reference:
+        //
+        // TypeError: 'clipboard-write' (value of 'name' member of
+        //   PermissionDescriptor) is not a valid value for enumeration
+        //   PermissionName.
+        //
+        if (error.message.includes("clipboard-write")) {
+          console.warn(error);
+          return false;
+        } else {
+          console.error(error);
+        }
+      });
   }
 }
```

If the error message is about "clipboard-write" specifically, we emit a warning instead of an error, because the "copy to clipboard" button will _probably_ work. Otherwise we emit an error and return `undefined` from that function. Which will also let the "copy to clipboard" button _probably_ work.